conf/distro/git_describe_version.inc

```diff
@@ -0,0 +1,13 @@
+def git_describe_version(repo_dir_path, default):
+    import subprocess
+
+    git_args = ['git', '-C', repo_dir_path, 'describe', '--tags', '--dirty']
+    ret = subprocess.run(git_args, stdout=subprocess.PIPE, universal_newlines=True)
+
+    if ret.returncode != 0:
+        bb.warn('Failed to run "git describe" in ' + repo_dir_path + ', using "' + default + '"')
+        return default
+
+    # Assumes there is a tag in the format "v<major>.<minor>/<Yocto release>".
```

Comment:
Supported tag format is:
"[v]<major>.<minor>[-e][</|->Yocto release name]"

Will support:
v1.0/pyro
v3.2-e-thud
3.2-e-thud
4.0-thud
4.0/thud
4.0

Reply:
Updated comment and moved it to top of file. Also added a copyright header.